diff --git a/src/common/constants/platform.ts b/src/common/constants/platform.ts
index 12721cb6f8965b71390eb79d207e029c5c61076f..1445ffc1a6a2dd6f482ce411c68cf5f664ca498a 100644
--- a/src/common/constants/platform.ts
+++ b/src/common/constants/platform.ts
@@ -10,12 +10,12 @@ function isWindowAvailable() {
 }
 
 export const IS_ANDROID: boolean = /* @__PURE__ */ (() =>
-  Platform.OS === 'android')();
-export const IS_IOS: boolean = /* @__PURE__ */ (() => Platform.OS === 'ios')();
-export const IS_WEB: boolean = Platform.OS === 'web';
+  Platform?.OS === 'android')();
+export const IS_IOS: boolean = /* @__PURE__ */ (() => Platform?.OS === 'ios')();
+export const IS_WEB: boolean = Platform?.OS === 'web';
 export const IS_JEST: boolean = !!process.env.JEST_WORKER_ID;
 /** @knipIgnore */
-export const IS_WINDOWS: boolean = Platform.OS === 'windows';
+export const IS_WINDOWS: boolean = Platform?.OS === 'windows';
 
 export const IS_WINDOW_AVAILABLE: boolean = isWindowAvailable();
 
